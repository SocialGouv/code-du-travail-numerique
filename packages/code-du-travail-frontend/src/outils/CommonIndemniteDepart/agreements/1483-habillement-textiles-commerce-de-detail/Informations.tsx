import { useContext } from "react";
import { Paragraph } from "@socialgouv/cdtn-ui";

import {
  StyledFilledElementSpan,
  StyledFilledElementTable,
} from "../../steps/Resultat/components/FilledElements";
import {
  IndemniteDepartContext,
  useIndemniteDepartStore,
} from "../../store";

export default function Agreement1483Informations() {
  const store = useContext(IndemniteDepartContext);
  const { noticeSalaryPeriods, hasReceivedSalaries } =
    useIndemniteDepartStore(store, (state) => ({
      noticeSalaryPeriods:
        state.agreement1483Data.input.noticeSalaryPeriods ?? [],
      hasReceivedSalaries: state.agreement1483Data.input.hasReceivedSalaries,
    }));

  return (
    <>
      {noticeSalaryPeriods.length > 0 && (
        <>
          <li>
            Connaissance du montant des salaires perçus pendant le préavis
            &nbsp;:&nbsp;{hasReceivedSalaries === "oui" ? "Oui" : "Non"}
          </li>

          {hasReceivedSalaries === "oui" && (
            <li>
              <Paragraph noMargin>
                Salaires perçus pendant le préavis&nbsp;:
              </Paragraph>
              <StyledFilledElementTable data-testid={"result-table"}>
                <thead>
                  <tr>
                    <th>Mois</th>
                    <th>
                      Salaire
                      <br />
                      <StyledFilledElementSpan>
                        (primes et avantages en nature inclus)
                      </StyledFilledElementSpan>
                    </th>
                    <th>Dont primes</th>
                  </tr>
                </thead>
                <tbody>
                  {noticeSalaryPeriods.map((salary, index) => (
                    <tr
                      key={"salary-agreement-" + index}
                      data-testid={"table-result-row"}
                    >
                      <td>{salary.month}</td>
                      <td>{salary.value} €</td>
                      <td>
                        {salary.prime} {salary.prime !== undefined && "€"}
                      </td>
                    </tr>
                  ))}
                </tbody>
              </StyledFilledElementTable>
            </li>
          )}
        </>
      )}
    </>
  );
}
