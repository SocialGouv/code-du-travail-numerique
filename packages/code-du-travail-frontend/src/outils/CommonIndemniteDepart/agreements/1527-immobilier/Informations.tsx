import { useContext } from "react";
import {
  IndemniteDepartContext,
  useIndemniteDepartStore,
} from "../../store";

export default function Agreement1527Informations() {
  const store = useContext(IndemniteDepartContext);
  const { hasCommission } = useIndemniteDepartStore(store, (state) => ({
    hasCommission: state.agreement1527Data.input.hasCommission,
  }));

  return (
    <>
      <li>
        Le contrat de travail pr√©voit le versement de commissions&nbsp;:&nbsp;
        {hasCommission === "oui" ? "Oui" : "Non"}
      </li>
    </>
  );
}
