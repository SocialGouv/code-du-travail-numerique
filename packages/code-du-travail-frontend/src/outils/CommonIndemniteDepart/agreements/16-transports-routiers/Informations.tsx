import { useContext } from "react";
import { IndemniteDepartContext, useIndemniteDepartStore } from "../../store";

export default function Agreement16Informations() {
  const store = useContext(IndemniteDepartContext);
  const { showVariablePay, hasVariablePay } = useIndemniteDepartStore(
    store,
    (state) => ({
      showVariablePay: state.agreement16Data.input.showVariablePay,
      hasVariablePay: state.agreement16Data.input.hasVariablePay,
    }),
  );

  return (
    <>
      {showVariablePay && (
        <li>
          Les salaires indiqu√©s comportent une partie variable &nbsp;:&nbsp;
          {hasVariablePay === "oui" ? "Oui" : "Non"}
        </li>
      )}
    </>
  );
}
