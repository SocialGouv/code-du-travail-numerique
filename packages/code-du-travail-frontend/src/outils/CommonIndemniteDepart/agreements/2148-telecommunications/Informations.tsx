import React, { useContext } from "react";
import { Paragraph } from "@socialgouv/cdtn-ui";
import {
  StyledFilledElementSpan,
  StyledFilledElementTable,
} from "../../steps/Resultat/components/FilledElements";
import { IndemniteDepartContext, useIndemniteDepartStore } from "../../store";

export default function Agreement2148Informations() {
  const store = useContext(IndemniteDepartContext);
  const { noticeSalaryPeriods, hasReceivedSalaries } = useIndemniteDepartStore(
    store,
    (state) => ({
      noticeSalaryPeriods:
        state.agreement2148Data.input.noticeSalaryPeriods ?? [],
      hasReceivedSalaries: state.agreement2148Data.input.hasReceivedSalaries,
    }),
  );

  return (
    <>
      {noticeSalaryPeriods.length > 0 && (
        <>
          <li>
            Connaissance du montant des salaires perçus pendant le préavis
            &nbsp;:&nbsp;{hasReceivedSalaries === "oui" ? "Oui" : "Non"}
          </li>

          {hasReceivedSalaries === "oui" && (
            <li>
              <Paragraph noMargin>
                Salaires perçus pendant le préavis&nbsp;:
              </Paragraph>

              <StyledFilledElementTable data-testid={"result-table"}>
                <thead>
                  <tr>
                    <th>Mois</th>
                    <th>
                      Salaire{noticeSalaryPeriods.length > 1 && "s"}
                      <br />
                      <StyledFilledElementSpan>
                        (primes et avantages en nature inclus)
                      </StyledFilledElementSpan>
                    </th>
                  </tr>
                </thead>
                <tbody>
                  {noticeSalaryPeriods.map((salary, index) => (
                    <tr
                      key={"salary-agreement-" + index}
                      data-testid={"table-result-row"}
                    >
                      <td>{salary.month}</td>
                      <td>{salary.value} €</td>
                    </tr>
                  ))}
                </tbody>
              </StyledFilledElementTable>
            </li>
          )}
        </>
      )}
    </>
  );
}
