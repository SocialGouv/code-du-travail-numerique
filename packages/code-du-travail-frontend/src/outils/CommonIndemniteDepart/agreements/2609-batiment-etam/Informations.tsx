import { useContext } from "react";
import {
  IndemniteDepartContext,
  useIndemniteDepartStore,
} from "../../store";

export default function Agreement2609Informations() {
  const store = useContext(IndemniteDepartContext);
  const { hasSameSalary, hasVariablePay } = useIndemniteDepartStore(
    store,
    (state) => ({
      hasSameSalary: state.salairesData.input.hasSameSalary,
      hasVariablePay: state.agreement2609Data.input.hasVariablePay,
    })
  );

  return (
    <>
      {hasSameSalary === "non" && (
        <li>
          Les salaires indiqu√©s comportent une partie variable &nbsp;:&nbsp;
          {hasVariablePay === "oui" ? "Oui" : "Non"}
        </li>
      )}
    </>
  );
}
