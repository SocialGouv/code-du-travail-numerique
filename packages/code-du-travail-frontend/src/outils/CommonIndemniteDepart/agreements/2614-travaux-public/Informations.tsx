import { useContext } from "react";
import { IndemniteDepartContext, useIndemniteDepartStore } from "../../store";

export default function Agreement2614Informations() {
  const store = useContext(IndemniteDepartContext);
  const { hasSameSalary, hasVariablePay } = useIndemniteDepartStore(
    store,
    (state) => ({
      hasSameSalary: state.salairesData.input.hasSameSalary,
      hasVariablePay: state.agreement2614Data.input.hasVariablePay,
    }),
  );

  return (
    <>
      {hasSameSalary === "non" && (
        <li>
          Les salaires indiqu√©s comportent une partie variable &nbsp;:&nbsp;
          {hasVariablePay === "oui" ? "Oui" : "Non"}
        </li>
      )}
    </>
  );
}
