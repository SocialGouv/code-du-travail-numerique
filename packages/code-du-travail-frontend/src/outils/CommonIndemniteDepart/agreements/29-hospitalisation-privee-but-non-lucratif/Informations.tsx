import { useContext, useEffect } from "react";
import {
  IndemniteDepartContext,
  useIndemniteDepartStore,
} from "../../store";

export default function Agreement29Informations() {
  const store = useContext(IndemniteDepartContext);
  const {
    hasSixBestSalaries,
    sixBestSalariesTotal,
    init,
    shouldAskSixBestSalaries,
  } = useIndemniteDepartStore(store, (state) => ({
    hasSixBestSalaries: state.agreement29Data.input.hasSixBestSalaries,
    sixBestSalariesTotal: state.agreement29Data.input.sixBestSalariesTotal,
    init: state.agreement29Function.init,
    shouldAskSixBestSalaries:
      state.agreement29Data.input.shouldAskSixBestSalaries,
  }));

  useEffect(() => {
    init();
  }, [init]);
  return shouldAskSixBestSalaries ? (
    <>
      <li>
        Connaissance de la somme des meilleurs salaires perçus sur une période
        de 6 mois consécutifs &nbsp;:&nbsp;
        {hasSixBestSalaries === "oui" ? "Oui" : "Non"}
      </li>
      {hasSixBestSalaries === "oui" && (
        <li>
          Somme des meilleurs salaires perçus sur une période de 6 mois
          consécutifs &nbsp;:&nbsp;
          {sixBestSalariesTotal} €
        </li>
      )}
    </>
  ) : (
    <></>
  );
}
