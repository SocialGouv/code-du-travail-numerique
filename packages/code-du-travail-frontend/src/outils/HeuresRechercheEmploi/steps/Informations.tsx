import data from "@cdt/data...simulateurs/heures-recherche-emploi.data.json";
import React from "react";

import Html from "../../../common/Html";
import { MatomoActionEvent, trackQuestion } from "../../../lib";
import { SelectQuestion } from "../../common/SelectQuestion";
import {
  filterSituations,
  getFormProps,
  getNextQuestionKey,
  getOptions,
  getPastQuestions,
  getSituationsFor,
} from "../../common/situations.utils";
import { SectionTitle } from "../../common/stepStyles";

const { questions, situations: allSituations } = data;

const criteriaOrder = questions.map(({ name }) => name);

const questionsMap = questions.reduce(
  (state, v) => ({ ...state, [v.name]: v }),
  {}
);

type OpenArray = Array<{
  key: string;
  status: boolean;
}>;

function StepInformations({ form }) {
  const { values } = form.getState();
  const { ccn, typeRupture, criteria = {} } = values;
  const idcc = ccn ? ccn.num : 0;

  const initialSituations = getSituationsFor(allSituations, {
    idcc,
    typeRupture,
  });

  const possibleSituations = filterSituations(initialSituations, criteria);

  const nextQuestionKey = getNextQuestionKey(
    possibleSituations,
    criteriaOrder,
    criteria
  );
  const nextQuestionOptions: any = getOptions(
    possibleSituations,
    nextQuestionKey
  );
  const pastQuestions = getPastQuestions(
    initialSituations,
    criteriaOrder,
    criteria
  );

  const [isOpenArray, setIsOpenArray] = React.useState<OpenArray>([]);

  const handleChange = (key: string) => {
    let isFound = false;
    const arr = isOpenArray.map((v) => {
      if (v.key === key) {
        isFound = true;
        return { ...v, status: !v.status };
      }
      return v;
    });
    if (!isFound) {
      setIsOpenArray([...isOpenArray, { key, status: true }]);
    } else {
      setIsOpenArray(arr);
    }
  };

  return (
    <>
      <SectionTitle>Statut du salari√©</SectionTitle>
      {pastQuestions.map(([key, answers]) => (
        <SelectQuestion
          key={key}
          name={`criteria.${key}`}
          options={answers}
          label={questionsMap[key].question}
          onChange={() => {
            trackQuestion(
              questionsMap[key].name,
              MatomoActionEvent.HEURE_RECHERCHE_EMPLOI,
              false
            );
            form.batch(() => {
              getFormProps({
                criteria,
                key,
                pastQuestions,
              }).forEach((key) => form.change(`criteria.${key}`, undefined));
            });
          }}
          tooltip={
            questionsMap[key].note !== undefined
              ? {
                  content: <Html>{questionsMap[key].note}</Html>,
                  trackableFn: (visibility) => {
                    if (visibility) {
                      trackQuestion(
                        questionsMap[key].name,
                        MatomoActionEvent.HEURE_RECHERCHE_EMPLOI
                      );
                    }
                  },
                }
              : undefined
          }
          isTooltipOpen={
            isOpenArray.find((v) => v.key === key)
              ? isOpenArray.find((v) => v.key === key)?.status
              : false
          }
          onSwitchTooltip={() => handleChange(key)}
        />
      ))}
      {nextQuestionKey &&
        nextQuestionOptions &&
        questionsMap[nextQuestionKey] &&
        questionsMap[nextQuestionKey].question && (
          <>
            <SelectQuestion
              name={`criteria.${nextQuestionKey}`}
              label={questionsMap[nextQuestionKey].question}
              options={nextQuestionOptions}
              tooltip={
                questionsMap[nextQuestionKey].note !== undefined
                  ? {
                      content: (
                        <Html>{questionsMap[nextQuestionKey].note}</Html>
                      ),
                      trackableFn: (visibility) => {
                        if (visibility) {
                          trackQuestion(
                            questionsMap[nextQuestionKey].name,
                            MatomoActionEvent.HEURE_RECHERCHE_EMPLOI
                          );
                        }
                      },
                    }
                  : undefined
              }
              onChange={() => {
                trackQuestion(
                  questionsMap[nextQuestionKey].name,
                  MatomoActionEvent.HEURE_RECHERCHE_EMPLOI,
                  false
                );
              }}
              isTooltipOpen={
                isOpenArray.find((v) => v.key === nextQuestionKey)
                  ? isOpenArray.find((v) => v.key === nextQuestionKey)?.status
                  : false
              }
              onSwitchTooltip={() => handleChange(nextQuestionKey)}
            />
          </>
        )}
    </>
  );
}

export { StepInformations };
