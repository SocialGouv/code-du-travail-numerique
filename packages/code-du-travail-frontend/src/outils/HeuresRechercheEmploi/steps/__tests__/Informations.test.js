import { renderForm } from "../../../../../test/renderForm";
import { StepInformations } from "../Informations";

describe("<StepInformations />", () => {
  it("should render", () => {
    const { container } = renderForm(StepInformations);
    expect(container).toMatchSnapshot();
  });
  it("should render catégorie professionelle", () => {
    const { container } = renderForm(StepInformations, {
      ccn: { selected: { num: 44 } },
      typeRupture: "1| Démission",
    });
    expect(container).toMatchSnapshot();
  });
  it("should render groupe", () => {
    const { container } = renderForm(StepInformations, {
      ccn: { selected: { num: 44 } },
      criteria: { "catégorie professionnelle": "28| Techniciens" },
      typeRupture: "1| démission",
    });
    expect(container).toMatchSnapshot();
  });
});
