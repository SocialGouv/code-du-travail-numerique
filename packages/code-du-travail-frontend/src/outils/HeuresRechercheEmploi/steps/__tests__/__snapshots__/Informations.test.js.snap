// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepInformations /> should render 1`] = `
.c0 {
  margin-top: 3.2rem;
  margin-bottom: 2rem;
  color: #4d73b8;
  font-weight: 600;
  font-size: 1.8rem;
  font-family: "Open Sans",sans-serif;
}

<div>
  <h2
    class="c0"
  >
    Statut du salarié
  </h2>
</div>
`;

exports[`<StepInformations /> should render catégorie professionelle 1`] = `
.c4 {
  color: #3e486e;
  line-height: 1.25;
  font-size: 1.6rem;
  font-weight: 400;
}

.c5 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  text-align: center;
}

.c8 {
  position: absolute;
  top: 1.6rem;
  right: 1.6rem;
  width: 1.6rem;
  height: 1.6rem;
  color: #f66663;
  pointer-events: none;
}

.c7 {
  width: 100%;
  height: 5.4rem;
  padding: 0 2rem 0;
  padding-right: 5rem;
  color: #3e486e;
  font-size: 1.6rem;
  font-family: "Open Sans",sans-serif;
  vertical-align: middle;
  background-color: #fff;
  border: none;
  border-radius: 0.6rem;
  box-shadow: 0 1rem 2rem rgba(121,148,212,0.2);
  cursor: pointer;
  -webkit-transition: border-color 250ms ease;
  transition: border-color 250ms ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c7::-ms-expand {
  background-color: transparent;
  border: 0 transparent;
}

.c7 *::-ms-backdrop,
.c7 {
  padding-right: 1.6rem;
}

.c7:invalid {
  border-color: #eb5757;
}

.c7:disabled {
  background-color: #e4e8ef;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 2rem;
  margin-bottom: 1rem;
  position: relative;
}

.c3 {
  font-size: 1.8rem;
  font-weight: 600;
  margin: 0;
}

.c6 {
  width: 40rem;
}

.c1 {
  margin-bottom: 3.2rem;
}

.c0 {
  margin-top: 3.2rem;
  margin-bottom: 2rem;
  color: #4d73b8;
  font-weight: 600;
  font-size: 1.8rem;
  font-family: "Open Sans",sans-serif;
}

@media (max-width:600px) {
  .c5 {
    width: 100%;
  }
}

@media (max-width:600px) {
  .c3 {
    font-size: 1.6rem;
  }
}

@media (max-width:600px) {
  .c6 {
    width: 100%;
  }
}

<div>
  <h2
    class="c0"
  >
    Statut du salarié
  </h2>
  <div
    class="c1"
  >
    <div>
      <div
        class="c2"
      >
        <label
          class="c3"
          for="input-criteria.catégorie professionnelle"
        >
          Quelle est la catégorie professionnelle du salarié ?
           
          <span
            class="c4"
          >
             (obligatoire)
          </span>
        </label>
      </div>
    </div>
    <div
      class="c5 c6"
    >
      <select
        class="c7"
        id="input-criteria.catégorie professionnelle"
        name="criteria.catégorie professionnelle"
      >
        <option
          disabled=""
          value=""
        >
          ...
        </option>
        <option
          value="6| Ouvriers, Employés"
        >
          Ouvriers, Employés
        </option>
        <option
          value="23| Agents de maîtrise"
        >
          Agents de maîtrise
        </option>
        <option
          value="28| Techniciens"
        >
          Techniciens
        </option>
        <option
          value="60| Ingénieurs, Cadres"
        >
          Ingénieurs, Cadres
        </option>
      </select>
      <div
        aria-hidden="true"
        class="c8"
      >
        <svg
          aria-hidden="true"
          fill="none"
          viewBox="0 0 16 16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            clip-path="url(#arrow-down_svg__clip0)"
          >
            <path
              d="M8.425 11.573l7.11-6.072a.709.709 0 00.26-.537.709.709 0 00-.26-.537.973.973 0 00-.629-.223.973.973 0 00-.63.223l-6.48 5.536-6.481-5.536a.973.973 0 00-.63-.223.973.973 0 00-.628.223.708.708 0 00-.261.537c0 .202.094.395.26.537l7.11 6.072a.91.91 0 00.29.165 1.021 1.021 0 00.68 0 .91.91 0 00.29-.165z"
              fill="currentColor"
            />
          </g>
          <defs>
            <clippath
              id="arrow-down_svg__clip0"
            >
              <path
                d="M0 0h16v16H0z"
                fill="currentColor"
              />
            </clippath>
          </defs>
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`<StepInformations /> should render groupe 1`] = `
.c0 {
  margin-top: 3.2rem;
  margin-bottom: 2rem;
  color: #4d73b8;
  font-weight: 600;
  font-size: 1.8rem;
  font-family: "Open Sans",sans-serif;
}

<div>
  <h2
    class="c0"
  >
    Statut du salarié
  </h2>
</div>
`;
